### PR TITLE
Feature/query and update

### PR DESCRIPTION
**Pull request by maths29**
In this project, I used transactions to update and delete records.I also use the knowledge of aggregate functions and GROUP BY to answer analytical questions.
Screenshot 2023-09-12 123043-required
Screenshot 2023-09-12 123516-required
Screenshot 2023-09-12 124333-required
Screenshot 2023-09-12 130511-required
Screenshot 2023-09-12 130756-required
Screenshot 2023-09-12 131925-required

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,69 @@
...
+
+SAVEPOINT SP2;
+
+UPDATE animals SET weight_kg=weight_kg * (-1);
+
+ROLLBACK TO SP1;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
```
Comment (mckent05) [inline queries.sql]: - Good job done so far, but you have not correctly run this transaction, please take your time to re-run all transactions and make sure you roll back to the correct savepoint. If your transactions were correctly run, you should have only 10 animals in your database. Please fix this 🙏

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,69 @@
...
+/*How many animals have never tried to escape?*/
+SELECT COUNT(*) from animals WHERE escape_attempts=0;
+/*What is the average weight of animals?*/
+SELECT AVG(weight_kg)::NUMERIC(10,2) As avg_weight from animals;
+/*Who escapes the most, neutered or not neutered animals?*/
+SELECT SUM(escape_attempts) AS escape_attempts,neutered from animals GROUP BY neutered;
+/*What is the minimum and maximum weight of each type of animal?*/
+SELECT MIN(weight_kg) AS minimum_weight,MAX(weight_kg) as max_weight,species from animals GROUP BY species;
+/*What is the average number of escape attempts per animal type of those born between 1990 and 2000?*/
+SELECT AVG(escape_attempts)::NUMERIC(10,2) from animals WHERE EXTRACT(YEAR FROM date_of_birth) BETWEEN 1990 AND 2000 GROUP BY species;
```
Comment (mckent05) [inline queries.sql:79]: - After running all transactions correctly, please attach the right screenshots for all your queries, Please fix this 🙏.

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @maths29, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You did well updating and querying your table 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @maths29, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**